contracts/poolV2.sol

```diff
@@ -238,13 +243,14 @@ contract Pool is iBEP20 {
             _amount = _getAddedBaseAmount();
             (outputAmount, fee) = _swapBaseToToken(_amount);
         }
-        emit Swapped(_fromToken, token, _amount, outputAmount, fee, member);
+        emit Swapped(_fromToken, token,member, _amount, outputAmount, fee);
         iBEP20(token).transfer(member, outputAmount);
         sync();
         return (outputAmount, fee);
     }
 
-    function mintSynths(address synthOut, address member) public onlyRouter returns(uint outputAmount, uint fee) {
+    function mintSynths(address synthOut, address member) external returns(uint outputAmount, uint fee) {
+      require(iSYNTHFACTORY(_DAO().SYNTHFACTORY()).isSynth(synthOut) == true, "!synth");
```

Comment:
Change from 'public' to 'external' to save gas.
Moved isSynth check to Pool level